ci/build_setup.sh

```diff
@@ -17,9 +17,9 @@ function setup_gcc_toolchain() {
 }
 
 function setup_clang_toolchain() {
```

Comment:
Per @lizan might as well just start linking w/ lld for the cases where we use clang today? Or prefer to defer to a separate PR?

Reply:
Let's defer that to a separate PR, IIRC we hardcoded `gold` in some bzl files.

Reply:
Definitely a separate PR. I believe that Bazel hardcodes `-fuse-ld=gold` somewhere as well, but maybe that's easy to override.

Reply:
It's pretty trivial using bazelrc but agree let's do in a follow up.